include/roboteam_ai/world/FieldMessage.h

```diff
@@ -14,124 +10,234 @@
+     * the field (where our goal is placed) and the right side of the field (where the opponents goal is placed). */
+    FIELD_LENGTH,
+    /* The width of both ours and the opponents goal (measured in meters, which is the y-coordinate difference between
+     * both goalposts). */
+    GOAL_WIDTH,
+    // The difference in x-coordinate (measured in meters) between the open part of the goal and the closed part of the goal.
+    GOAL_DEPTH,
+    BOUNDARY_WIDTH, // The width (measured in meters) of the boundary around the field.
+    CENTER_Y, // The center y-coordinate of the field (the y-coordinate that corresponds with the center of the field)
+    LEFTMOST_X, //The leftmost x-coordinate of the field (the x-coordinate closest to our goal)
+    RIGHTMOST_X, //The rightmost x-coordinate of the field (the x-coordinate closest to the opponents goal)
+    BOTTOMMOST_Y, //The bottommost y-coordinate of the field (the y-coordinate corresponding to the bottom side of the field)
+    TOPMOST_Y, //The uppermost y-coordinate of the field (the y-coordinate corresponding to the upper side of the field)
+};
+
+enum FieldLineName {
+    TOP_LINE, // The field line with the highest y-coordinate which goes from the left side to the right side of the field.
+    BOTTOM_LINE, // The field line with the lowest y-coordinate which goes from the left side to the right side of the field.
+    LEFT_LINE, // The field line left from our goal (our goal is always adjacent to this line).
+    RIGHT_LINE, // The field line right from the opponents goal (their goal is always adjacent to this line)
+    /* The line that seperates our side from the field and the opponent side of the field (this line moves in the y
+     * direction of the field, i.e. the width of the field) */
+    HALF_LINE,
+    /* The line that start from the middle point of our goal and ends at the middle point of the opponents goal (this
+     * line moves in the x direction of the field, i.e. the length of the field) */
+    CENTER_LINE,
+    LEFT_PENALTY_LINE, // The line parallel to the left line and our goal. This line marks our goal area.
+    RIGHT_PENALTY_LINE, // The line parallel to the right line and the opponents goal. This line marks their goal area.
+    /* The line closest (of all our goal area lines) to the top line and parallel to the top line. This line marks our
+     * goal area. */
+    TOP_LEFT_PENALTY_STRETCH,
+    /* The line closest (of all our goal area lines) to the bottom line and parallel to the bottom line. This line marks
+     * our goal area. */
+    BOTTOM_LEFT_PENALTY_STRETCH,
+    /* The line closest (of all the opponents goal area lines) to the top line and parallel to the top line. This line
+     * marks the opponents goal area. */
+    TOP_RIGHT_PENALTY_STRETCH,
+    /* The line closest (of all the opponents goal area lines) to the bottom line and parallel to the bottom line. This
+     * line marks the opponents goal area. */
+    BOTTOM_RIGHT_PENALTY_STRETCH
+};
```

Comment:
Please use enum classes

Reply:
They prevent implicit conversion (e.g. if you would now call field.get(0) you have VERY undefined behavior as it can be interpreted as the first element of any of these 4 enums.

Reply:
I do not agree, because for enum classes you need to write FieldLineName::RIGHT_LINE everytime you call the getter for the RIGHT_LINE.